aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/UpdateHandler.java

```diff
@@ -0,0 +1,183 @@
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import com.amazonaws.util.CollectionUtils;
+import software.amazon.awssdk.services.rds.RdsClient;
+import software.amazon.awssdk.services.rds.model.DBInstance;
+import software.amazon.awssdk.services.rds.model.DbInstanceNotFoundException;
+import software.amazon.awssdk.services.rds.model.DbInstanceRoleAlreadyExistsException;
+import software.amazon.awssdk.services.rds.model.DbInstanceRoleNotFoundException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class UpdateHandler extends BaseHandlerStd {
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<RdsClient> proxyClient,
+            final Logger logger
+    ) {
+
+        final ResourceModel previousModel = request.getPreviousResourceState();
+        final ResourceModel desiredModel = request.getDesiredResourceState();
+
+        final Collection<Tag> desiredTags = Translator.translateTagsFromRequest(request.getDesiredResourceTags());
+
+        final Collection<DBInstanceRole> previousRoles = previousModel.getAssociatedRoles();
+        final Collection<DBInstanceRole> desiredRoles = desiredModel.getAssociatedRoles();
+
+        return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
+                .then(progress -> {
+                    if (progress.getCallbackContext().isModified()) {
+                        return progress;
+                    }
+
+                    return proxy.initiate("rds::modify-db-instance", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+                            .translateToServiceRequest(resourceModel -> Translator.modifyDbInstanceRequest(previousModel, desiredModel))
+                            .backoffDelay(BACKOFF_STRATEGY)
+                            .makeServiceCall((modifyRequest, proxyInvocation) -> {
+                                return proxyInvocation.injectCredentialsAndInvokeV2(
+                                        modifyRequest,
+                                        proxyInvocation.client()::modifyDBInstance
+                                );
+                            })
+                            .done((modifyRequest, modifyResponse, proxyInvocation, resourceModel, context) -> {
+                                context.setModified(true);
```

Comment:
No need to update the context if the test suite does not use this transition assessment